### PR TITLE
Allow setting user data on tracks

### PR DESCRIPTION
**Pull request by topi314**
https://github.com/lavalink-devs/Lavalink/pull/984

### CODE REVIEW

File: protocol/src/commonMain/kotlin/dev/arbjerg/lavalink/protocol/v4/player.kt [hunk 1]
```diff
@@ -23,8 +23,13 @@ data class Player(
 data class Track(
     val encoded: String,
     val info: TrackInfo,
-    val pluginInfo: JsonObject
-) : LoadResult.Data
+    val pluginInfo: JsonObject,
+    val userData: JsonObject
+) : LoadResult.Data {
+    fun copyWithUserData(userData: JsonObject): Track {
```
Comment (duncte123) [inline protocol/src/commonMain/kotlin/dev/arbjerg/lavalink/protocol/v4/player.kt:67]: Maybe it's a good idea to add "deserializeUserData" like was done in #973 for plugin info

File: protocol/src/commonTest/kotlin/PlayerSerializerTest.kt [hunk 2]
```diff
@@ -127,7 +129,7 @@ class PlayerSerializerTest {
     @JsName("test3")
     fun `test encodedTrack and identifier exclusivity`() {
         //language=json
-        val json = """{"encodedTrack":  "", "identifier":  ""}"""
+        val json = """{"track": {"encodedTrack":  "", "identifier":  ""}}"""
```
Comment (duncte123) [inline protocol/src/commonTest/kotlin/PlayerSerializerTest.kt]: Invalid json property
```suggestion
        val json = """{"track": {"encoded":  "", "identifier":  ""}}"""
```

File: LavalinkServer/src/main/java/lavalink/server/player/PlayerRestHandler.kt [hunk 3]
```diff
@@ -59,8 +59,24 @@ class PlayerRestHandler(
     ): ResponseEntity<Player> {
         val context = socketContext(socketServer, sessionId)
 
-        val encodedTrack = playerUpdate.encodedTrack
-        if (encodedTrack is Omissible.Present && playerUpdate.identifier is Omissible.Present) {
+        val track = if (playerUpdate.track.isPresent()) {
+            playerUpdate.track
+        } else {
```
Comment (freyacodes) [inline LavalinkServer/src/main/java/lavalink/server/player/PlayerRestHandler.kt:68]: We should return 400 Bad Request if both methods of specification are used

**Review by freyacodes — CHANGES_REQUESTED** [review]
Documentation is needed

**Comment by topi314** [comment]
docs are here https://github.com/lavalink-devs/Lavalink/pull/984

**Review by freyacodes — APPROVED** [review]

**Review by duncte123 — APPROVED** [review]

**Pull request merged**